openhtf/util/logs.py

```diff
@@ -152,11 +152,14 @@ def __init__(self):
   def filter(self, record):
     if self.MAC_REPLACE_RE.search(record.getMessage()):
       # Update all the things to have no mac address in them
-      record.msg = self.MAC_REPLACE_RE.sub(self.MAC_REPLACEMENT, record.msg)
-      record.args = tuple([
-          self.MAC_REPLACE_RE.sub(self.MAC_REPLACEMENT, str(arg))
-          if isinstance(arg, six.string_types)
-          else arg for arg in record.args])
+      if isinstance(record.msg, six.string_types):
+        record.msg = self.MAC_REPLACE_RE.sub(self.MAC_REPLACEMENT, record.msg)
+        record.args = tuple([
+            self.MAC_REPLACE_RE.sub(self.MAC_REPLACEMENT, str(arg))
+            if isinstance(arg, six.string_types)
+            else arg for arg in record.args])
+      else:
+        record.msg = self.MAC_REPLACE_RE.sub(self.MAC_REPLACEMENT, record.getMessage())
```

Comment:
Works under the assumption that record.args is empty if record.msg is not a string type. Strictly speaking not necessarily true, but should be close enough.

Reply:
record.getMessage() does the substituting of the args if they exist, that's why on line 153 it will catch cases regardless of whether the mac-address is in record.msg or record.args.

Reply:
It's pretty good, but consider this (somewhat pathological) edge-case that lets a MAC address through:

```
class StringHolder(object):
  def __init__(self, s):
    self.s = s
  def __str__(self):
    return self.s

leaked_mac = '00:14:22:01:23:45'
logging.warn(StringHolder('%%s%s'), leaked_mac)
```

I don't care too much about this since there are pre-existing, easier ways for MAC addresses to sneak through, e.g.:

```
logging.warn('%s', StringHolder(leaked_mac))
```

Reply:
I guess what's more relevant is I thought this would break cases like

```python
logging.warn(StringHolder('%s'), 'hi')
```

But it looks like that's not the case